.github/workflows/build.yml

```diff
@@ -4,99 +4,105 @@ on: [push, pull_request]
+            os: ubuntu-latest
+          - php: 7.2
+            coverage: none
+            extensions: apc, curl, dom, imagick, intl, mbstring, mcrypt, memcached
+            os: ubuntu-latest
+          - php: 7.3
+            coverage: none
+            extensions: apc, curl, dom, imagick, intl, mbstring, mcrypt, memcached
+            os: ubuntu-latest
+          - php: 7.4
+            coverage: xdebug
+            extensions: apc, curl, dom, imagick, intl, mbstring, mcrypt, memcached
+            os: ubuntu-latest
+          - php: 8.0
+            coverage: none
+            extensions: apc, curl, dom, imagick, intl, mbstring, mcrypt, memcached
+            os: ubuntu-latest
+          - php: 8.1
+            coverage: none
+            extensions: apcu, curl, dom, imagick, intl, mbstring, mcrypt, memcached
+            os: ubuntu-latest
+          - php: 8.2
+            extensions: apcu, curl, dom, imagick, intl, mbstring, mcrypt, memcached
+            coverage: none
+            os: ubuntu-latest
 
     steps:
-      - name: Generate french locale
+      - name: Generate french locale.
         run: sudo locale-gen fr_FR.UTF-8
-      - name: Checkout
+
+      - name: Checkout.
         uses: actions/checkout@v3
-      - name: Install PHP
+
+      - name: Install PHP.
         uses: shivammathur/setup-php@v2
         with:
+          coverage: ${{ matrix.coverage }}
+          extensions: ${{ matrix.extensions }}
+          ini-values: apc.enabled=1,apc.shm_size=32M,apc.enable_cli=1, date.timezone='UTC', session.save_path="${{ runner.temp }}"
           php-version: ${{ matrix.php }}
           tools: pecl
-          extensions: apc, curl, dom, imagick, intl, mbstring, mcrypt, memcached, mysql, pdo, pdo_mysql, pdo_pgsql, pdo_sqlite, pgsql, sqlite
-          ini-values: date.timezone='UTC', session.save_path="${{ runner.temp }}"
-      - name: Install Memcached
+
+      - name: Install Memcached.
         uses: niden/actions-memcached@v7
-      - name: Get composer cache directory
-        id: composer-cache
-        run: echo "dir=$(composer config cache-files-dir)" >> $GITHUB_OUTPUT
-      - name: Cache composer dependencies
-        uses: actions/cache@v3
-        with:
-          path: ${{ steps.composer-cache.outputs.dir }}
-          key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.lock') }}
-          restore-keys: ${{ runner.os }}-composer-
-      - name: Install dependencies
+
+      - name: Install dependencies.
         run: composer update $DEFAULT_COMPOSER_FLAGS
-      - name: PHP Unit tests
+
+      - name: Run tests with phpunit.
+        if: matrix.php < '7.4' || matrix.php >= '8.1'
         run: vendor/bin/phpunit --verbose --exclude-group $PHPUNIT_EXCLUDE_GROUP --colors=always
 
-  npm:
-    name: NPM 6 on ubuntu-latest
-    runs-on: ubuntu-latest
+      - name: Run tests with phpunit.
+        if: matrix.php == '8.0'
+        continue-on-error: true
```

Comment:
That would result in "green" tests even if there are failures, right? I'd like to avoid that.

Reply:
Well in php 8.0 the tests are segmented having passed all the tests correctly, so either we add the continue-on: error, or we separate the 8.0 tests from the compilation, or we simply leave them as failure.

Reply:
Done, problem solved, now the tests are not segmented.